src/vs/vscode.d.ts

```diff
@@ -4849,16 +4889,19 @@ declare module 'vscode' {
 		export const onDidSaveTextDocument: Event<TextDocument>;
 
 		/**
-		 * Get a configuration object.
+		 * Get a workspace configuration object.
 		 *
 		 * When a section-identifier is provided only that part of the configuration
 		 * is returned. Dots in the section-identifier are interpreted as child-access,
 		 * like `{ myExt: { setting: { doIt: true }}}` and `getConfiguration('myExt.setting').get('doIt') === true`.
 		 *
+		 * When a resource is provided, configuration scoped to that resource is returned.
+		 *
 		 * @param section A dot-separated identifier.
-		 * @return The full workspace configuration or a subset.
+		 * @param resource A resource for which the configuration is asked for
+		 * @return The full configuration or a subset.
 		 */
-		export function getConfiguration(section?: string): WorkspaceConfiguration;
+		export function getConfiguration(section?: string, resource?: Uri): WorkspaceConfiguration;
```

Comment:
This is a little asymmetric. I like the idea of having a `resource` because it comes close to my much wished editorconfig-style model but we have `folderValue` from `inspect` and we talk about workspace folders in the jsdoc of `WorkspaceConfiguration`. We should decide for one, so either always resource or workspace folder